cloudinit/config/cc_puppet.py

```diff
@@ -118,26 +119,27 @@ def __init__(
         self.csr_attributes_path = csr_attributes_path
 
 
-def _autostart_puppet(log):
+def _autostart_puppet(log, cloud: Cloud, service_name: str):
     # Set puppet to automatically start
-    if os.path.exists("/etc/default/puppet"):
+    if os.path.exists(f"/etc/default/{service_name}"):
```

Comment:
shouldn't this be using `distro.manage_service('enable', service_name)`, or is there a reason you're avoiding that?

Reply:
@igalic, no specific reason. I am still improving the PR. Thank you for the suggestion.